CIPs/cip-145.md

```diff
@@ -51,17 +56,15 @@ branches are pruned. Since pruning branches that contain Data Events would resul
 a Data Event to contain multiple ancestor events in its `prev` field so that a new merge point Data Event can cover
 multiple events.
 
-1. If a stream is in a diverged state (see events `A` and `B` in fig. 5), we only consider branches that contain valid,
-   uncovered Data Events.
-2. For branches that contain valid, uncovered Data Events, we only consider the first Data Event on a branch after the
-   fork.
-3. The Data Event that is covered by the earliest Time Event wins (see event `A` in fig. 5). 
-4. If two uncovered Data Events are covered by Time Events at the same block height, the Data Event with the lower CID
-   wins.
-
-Using multi-prev Data Events allows us to reduce the number of uncovered events and converge the stream so that there is
-only a single uncovered event, without any data abandoned on pruned branches. The stream's converged/diverged state can
-be determined by looking at the `prev` fields of all the Data Events for that stream.
+1. If a stream is in a diverged state (see events `A` and `B` in fig. 5), we consider branches that contain dominant
+   Data Events. Branches containing only Time Events will get pruned.
+2. For branches that contain dominant Data Events, consider the first Data Event on a branch after the fork.
+3. The Data Event that is covered by the earliest Time Event wins (see event `A` in fig. 5).
+4. If two Data Events share the earliest timestamp, then the branch of the Data Event with the lower CID wins.
```

Comment:
What does it mean for a Data Event to "win" in the context of multiple prev?

Reply:
We updated the wording. A Data Event "winning" here meant that that Data Event would become the tip.

Reply:
Ok I don't understand why we need to distinguish between winning and non-winning tips? Why not just call them all tips and put them all in the `prev` field? I don't follow why the protocol needs to care about "winning"?

Reply:
Yes, when I say protocol here I'm referring to the event streaming protocol. Stream type handlers is an application.

Reply:
Makes full-on sense then :ok_hand:

Reply:
> Ok I don't understand why we need to distinguish between winning and non-winning tips? Why not just call them all tips and put them all in the `prev` field? I don't follow why the protocol needs to care about "winning"?
> e.g. The protocol just gives the application a list of tips. It's up to the application to decide what is "winning" and what's not. It's also up to the application to chose the order of tips in it's `prev` array when it's being constructed.

To answer your question, @oed, there are two reasons for this:
- It is simpler for applications to just be given a tip per some default, predictable algorithm. They can choose to override this order, but don't have to.
- There is always _some_ eventually consistent state of a diverged stream, even if the controller never comes back to create the Merge Event, because the default precedence rules can be used to determine the tip.

Reply:
Wouldn't the application need to be given all tips anyway? In order to include them all in the `prev` array? Are we simply talking about the ordering of the CIDs in the returned array here?